src/main/scala/com/campudus/tableaux/Starter.scala

```diff
@@ -47,39 +47,31 @@ class Starter extends Verticle {
 
       _ <- deployHttpServer(port, tableauxConfig)
     } yield {
-      p.success()
+      p.success(())
     }
   }
 
   def createUploadsDirectory(config: TableauxConfig): Future[Unit] = promisify { p: Promise[Unit] =>
     val uploadsDirectory = Path(s"${config.workingDirectory}/${config.uploadsDirectory}")
 
-    vertx.fileSystem.mkdir(s"$uploadsDirectory", { asyncResult =>
-      p.success()
-    })
+    // succeed also in error cause (directory already exists)
+    vertx.fileSystem.mkdir(s"$uploadsDirectory", { result => p.success() })
```

Comment:
You might not have all permissions to create this directory, so it's not always only "directory already exists". Maybe it's a special exception (EEXISTS) that can be checked against and succeed in that special case.